### ISSUE TITLE
data augmentation error

### ISSUE THREAD

**Issue by LinlyAC** [fizyr/keras-retinanet#227]
I found that you have updated about data augmentation.
I made this data augmentation in my code for finetuning a new model based on your best model Resnet50,

![1](https://user-images.githubusercontent.com/16618172/34981103-78086360-fae1-11e7-839f-dd9668a17fc3.png)

but the loss is werid:

![3](https://user-images.githubusercontent.com/16618172/34982207-bd19955c-fae4-11e7-9cc0-45ae37eedc50.png)

the classfcation loss is very high but the regression loss is almost to zero. Is there something wrong?

**Comment by de-vri-es** [comment]
Rotation is in radians, so 40 radians is quite a few full circles. Translation is relative to the width/height of the image, so max translation is ten times the width/height of your images. That easily puts most of your images completely out of the image canvas. I expect that could very well explain bad training results.

Could you try with some lower values?

**Comment by hgaiser** [comment]
Should we add a check for the translation so that it is between [0, 1) ? No sense in other values I suppose?

**Comment by LinlyAC** [comment]
@de-vri-es so what you say is my rotation is small and the translation is too big?

**Comment by de-vri-es** [comment]
> @de-vri-es so what you say is my rotation is small and the translation is too big?

Actually, they're both very big. I assume you meant 40 degrees which is only 0.698 radians. Unless you have a reason not to, I'd also center the rotation around 0, so set min to -0.35 and max to 0.35 or something along those lines.

> Should we add a check for the translation so that it is between [0, 1) ? No sense in other values I suppose?

The only problem with that is that it is the data generator that interprets the translation as factor of the image size. The `random_transform_generator`  truly puts values in the range of whatever you give. Limiting that to `[0, 1)` makes it much less generic.

We could also simply make the translation absolute in pixels, instead of interpreting it as factor of the image size. Or at the very least document it

**Comment by LinlyAC** [comment]
thank you very much!

**Comment by LinlyAC** [comment]
another question, if I use the new augmentation way, should I adjust some optimizer parameters, such as adam lr?

**Comment by de-vri-es** [comment]
I submitted a PR (#228) to let the user choose absolute translations, and to improve the documentation. Hopefully that will make it more clear what values make sense here. 

@LinlyAC: Could you have a look at #228 if the updated documentation is clear enough according to you?

> another question, if I use the new augmentation way, should I adjust some optimizer parameters, such as adam lr?

To be honest, I don't know. Maybe someone else can shed some light on that. The best place for that would be the slack channel though (see the README.md for details).

**Comment by LinlyAC** [comment]
It is very useful, thank you!

**Comment by hgaiser** [comment]
I guess this is resolved then?

**Comment by de-vri-es** [comment]
For the record: #228 is merged, so absolute translations are possible now
